controllers/agentmachine_controller.go

```diff
@@ -115,7 +117,7 @@ func (r *AgentMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 
 	if agentCluster.Status.ClusterDeploymentRef.Name == "" {
-		return r.updateStatus(ctx, log, agentMachine, ctrl.Result{}, fmt.Errorf("No cluster deployment reference on agentCluster %s", agentCluster.GetName()))
+		return r.updateStatus(ctx, log, agentMachine, ctrl.Result{RequeueAfter: agentMachineDependenciesWaitTime}, nil)
```

Comment:
Hmmm, I think this state is invalid and should be an error, 
If not an error it deserves at least a warning log (otherwise we will have no idea why the AgentMachine is stuck)

Reply:
You're right, this should be resolved by the time machines start coming up